uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js

```diff
@@ -1,227 +1,221 @@
-                $log.debug(data);
-
-                var obj = $rootScope.cache.assignment.obj || {}, arr = [], arrSearch = [], assignment, assignmentId,
-                    assignmentType;
-
-                angular.extend(obj,data);
-
-                for (assignmentId in obj) {
-                    assignment = obj[assignmentId];
-                    assignmentType = $rootScope.cache.assignmentType.obj[assignment.case_type_id].title;
-                    arr.push(assignment);
-
-                    arrSearch.push({
-                        label: assignment.contacts[0].sort_name + ' - ' + assignmentType + (assignment.end_date ? ' (closed)' : ''),
-                        label_class: +assignment.is_deleted || assignment.end_date ? 'strikethrough' : '',
-                        id: assignmentId,
-                        extra: {
-                            case_subject: assignment.subject,
-                            case_type: assignmentType,
-                            contact_id: assignment.contacts[0].contact_id,
-                            end_date: assignment.end_date,
-                            sort_name: assignment.contacts[0].sort_name,
-                            start_date: assignment.start_date
-                        }
-                    });
-
-                }
-
-                arr = $filter('orderBy')(arr, 'subject');
-                arrSearch = $filter('orderBy')(arrSearch, 'label');
-
-                $rootScope.cache.assignment = {
-                    arr: arr,
-                    obj: obj,
-                    arrSearch: arrSearch
-                };
-            }
-        }
 
-    }]);
+            deferred.resolve(data.values);
+          }, function () {
+            deferred.reject('Unable to fetch assignment types');
+          });
+
+          return deferred.promise;
+        },
+        save: function (assignment) {
+          if (!assignment || typeof assignment !== 'object') {
+            return null;
+          }
+
+          var deferred = $q.defer();
+          var params = angular.extend({
+            sequential: 1,
+            debug: config.DEBUG
+          }, assignment);
+          var val;
+
+          Assignment.save({
+            action: 'create',
+            json: params
+          }, null, function (data) {
+            if (UtilsService.errorHandler(data, 'Unable to save an assignment', deferred)) {
+              return;
+            }
 
+            val = data.values;
+            deferred.resolve(val.length === 1 ? val[0] : null);
+          }, function () {
+            deferred.reject('Unable to save an assignment');
+          });
+
+          return deferred.promise;
+        },
+        search: function (input, excludeCaseId, includeContactIds) {
+          return AssignmentSearch.query({
+            sortName: input,
+            excludeCaseIds: excludeCaseId,
+            includeContactIds: Array.isArray(includeContactIds) ? includeContactIds.join(',') : includeContactIds
```

Comment:
I would suggest this, this would make sure that no error happens if any other value is sent.
```
search: function (input, excludeCaseId, includeContactIds) {
         // Check if includeContactIds is array, else assign to default []
         includeContactIds = Array.isArray(includeContactIds) ? includeContactIds : [];
          return AssignmentSearch.query({
            sortName: input,
            excludeCaseIds: excludeCaseId,
            includeContactIds: includeContactIds
          }).$promise;
        },
```